aptos-move/framework/aptos-framework/sources/account/account.move

```diff
@@ -489,6 +493,84 @@ module aptos_framework::account {
         });
     }
 
+    /// Upserts an ED25519 backup key to a keyless account by converting the account's authentication key to a multi-key.
+    /// This function takes a keyless account (identified by having a keyless public key is the original public key) and updates the
+    /// account's authentication key to a multi-key of the original keyless public key and the new backup key that requires
+    /// 1 signature from either key to authenticate.
+    ///
+    /// Note: This function emits a `KeyRotationToMultiPublicKey` event marking both keys as verified since the keyless public key
+    /// is the original public key of the account and the new backup key has been validated via verifying the challenge signed by the new backup key.
+    ///
+    /// # Arguments
+    /// * `account` - The signer representing the keyless account
+    /// * `keyless_public_key` - The original keyless public key of the account (wrapped in an AnyPublicKey)
+    /// * `backup_public_key` - The ED25519 public key to add as a backup
+    /// * `backup_key_proof` - A signature from the backup key proving ownership
+    ///
+    /// # Aborts
+    /// * If the any of inputs deserialize incorrectly
+    /// * If the provided public key is not a keyless public key
+    /// * If the keyless public key is not the original public key of the account
+    /// * If the backup key proof signature is invalid
+    ///
+    /// # Events
+    /// * Emits a `KeyRotationToMultiPublicKey` event with the new multi-key configuration
+    entry fun upsert_ed25519_backup_key_on_keyless_account(account: &signer, keyless_public_key: vector<u8>, backup_public_key: vector<u8>, backup_key_proof: vector<u8>) acquires Account {
+        // Check that the provided public key is a keyless public key
+        let keyless_single_key = single_key::new_public_key_from_bytes(keyless_public_key);
+        assert!(single_key::is_keyless_or_federated_keyless_public_key(&keyless_single_key), error::invalid_argument(ENOT_A_KEYLESS_PUBLIC_KEY));
+
+        let addr = signer::address_of(account);
+        let account_resource = &mut Account[addr];
+        let old_auth_key = account_resource.authentication_key;
+
+        // Check that the provided public key is original public key of the account by comparing
+        // its authentication key to the account address.
+        assert!(
+            bcs::to_bytes(&addr) == keyless_single_key.to_authentication_key(),
+            error::invalid_argument(ENOT_THE_ORIGINAL_PUBLIC_KEY)
+        );
```

Comment:
why original public key? what if the account has been rotated to a normal account, it's not a keyless account anymore.

Reply:
Then it does not match your description saying that "Upserts an ED25519 backup key to a keyless account" since it is not a keyless account anymore.
please refactor this part of the logic to make sure the account is a keyless account. I guess you just need to verify the auth_key and it has to equal to the address. I guess now we don't support an account rotating to keyless account, right?

Reply:
well i will just update the description to reflect that the requirement that the account is originally a keyless account.  

the current public key does not have to be keyless if it is rotated.

i'd prefer not to name it `upsert_ed25519_backup_key_on_originally_keyless_account`

Reply:
The target use case is very specific and the name is confusing. you only allow an original keyless account to add a backup key even now it is not a keyless account but the name applies only to real keyless account.

I think it only make sense:
either, you only allow real keyless account and abort if the address and auth_key does not match.
or, you make this function more general that it allows any single key account to add another key for backup since your event is also a general one.